library/java/net/openid/appauth/BrowserNotSupportedException.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018 The AppAuth for Android Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.openid.appauth;
+
+import net.openid.appauth.browser.BrowserDescriptor;
+
+/**
+ * Returned as a response to prepareAuthorizationRequestIntent if no supported browser is installed.
+ */
+public final class BrowserNotSupportedException extends RuntimeException {
```

Comment:
I think my only complaint about this change is that it will break existing code that is currently doing a try/catch for ActivityNotFoundException. A compromise that preserves backwards compatibility would be for `BrowserNotSupportedException` to extend from `ActivityNotFoundException`, and in doing so, I would probably rename this class to `BrowserNotFoundException` to fit the semantic pattern.

What do you think?

Reply:
@hnljp ping on this - do you agree with me, or do you have an alternative approach you would like to discuss?

Reply:
Sorry for the long delay.
I think that it is better just to stay with ActivityNotFoundException as it is for now.

Reply:
OK, no problem, I'll close this PR.